ethereum/block/testdata/UTContract.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: LGPLv3
+pragma solidity ^0.8.5;
+
+//
+// Unit test contract. Our goal is a transaction that is easy to make fail, but only when it is
+// mined, not when the transaction is sent. To do this, pass check_stamp a value that is equal
+// to the timestamp of the last mined block. Gas estimation will pass, as it uses the timestamp
+// from the last mined block, so we'll successfully send the transaction to the nework, but the
+// transaction is guaranteed to fail in whatever block it is mined into, as the block's timestamp
+// will be greater than what we passed in.
+//
+contract UTContract {
+    uint256 private stamp;
+
+    function check_stamp(uint256 _stamp) external {
+        require(block.timestamp <= _stamp, "block.timestamp was not less than stamp");
+        stamp = _stamp; // Prevent the function from being view-only
+    }
+}
```

Comment:
This is the contract I used when developing `errorFromBlock` and it seemed like the best way to unit test it. Even if we could test it using the `SwapFactory` contract, that package depends on this one, so we couldn't use it from here. Go tools ignore folders named `testdata`. Normally, people place input data for Unit tests in these directories, but it was the best idea I could come up with for where to put data used to generate code that test cases link against.